samples/net/gsm_modem/src/main.c

```diff
@@ -93,13 +93,14 @@ static void event_handler(struct net_mgmt_event_callback *cb,
 int main(void)
 {
 	const struct device *uart_dev =
-				device_get_binding(CONFIG_MODEM_GSM_UART_NAME);
+		DEVICE_DT_GET(DT_BUS(DT_INST(0, zephyr_gsm_ppp)));
 
 	gsm_dev = device_get_binding(GSM_MODEM_DEVICE_NAME);
```

Comment:
Shoudn't we convert this device_get_binding?

Reply:
I think we should, I tried:

```c
gsm_dev = DEVICE_DT_GET(DT_INST(0, zephyr_gsm_ppp));
```

but it fails to compile:
```
undefined reference to `__device_dts_ord_274'
```

Any idea what I missed?

Reply:
`DEVICE_DEFINE` should be replaced by `DEVICE_DEFINE_DT` in `gsm_ppp.c` and `GSM_MODEM_DEVICE_NAME` should be dropped.